README.md

```diff
@@ -50,35 +50,36 @@ pip install git+https://github.com/gridsingularity/d3a-api-client.git
 
 ## How to use the Client
 
-### Interacting with the CLI
-You can run one of the setup scripts residing in the `d3a_api_client/setups` folder, or you can 
-set the path for your scripts directory as list in the `d3a_api_client.constants` .
-In order to run a module you can use the following command
-```
-d3a-api-client run --setup {module_name}
-```
+### Interacting via CLI
+In order to get help, please run:
+```
+d3a-api-client run --help
+```
+Following is the more in depth explanation of individual flags supported,
+- base-setup-path --> Sets the base path where you client script resides else `d3a_api_client/setups` is taken as default for your client-scripts. User could provided both an absolute and relative file path.
+- setup --> Name of your api-client module/script.
+- username --> Username of agent authorized to communicate with respective collaboration/CN.
+- password --> Password of respective agent
+- domain-name --> D3A domain name
+- web-socket --> D3A websocket URI
+- simulation-config-path --> Path of JSON format file that contains agent's point of interest's collaboration/CN's info (It could downloaded from the registry page of UI) 
+- run-on-redis --> This flag targets the local testing of api client, where no user authentication. A locally running redis server d3a simulation is needed here.
 
-### Authentication for REST API
-Authentication is done implicitly. The d3a-client is reading the following two variable from the environment
-and requests a auth token from the D3A API which will be reused internally for all communication.
-```
-API_CLIENT_USERNAME
-API_CLIENT_PASSWORD
+#### Examples
+- For local testing of api-client
 ```
-The user credentials can be provided in two ways:
-#### Via CLI
-Users can use the following parameters in the `d3a_api_client` cli for setting username and password
-```
-  -u, --username TEXT     D3A username
-  -p, --password TEXT     D3A password
+d3a-api-client --log-level ERROR run --setup test_redis_aggregator --run-on-redis
 ```
+- For testing your api client script on remote server hosting d3a's collaboration/CNs.
+    - If your client script resides on `d3a_api_client/setups`
+    ```
+    d3a-api-client run -u <username> -p <password> --setup test_create_aggregator --simulation-config-path <your-downloaded-simulation-config-file-path>
+    ```
+    - If your client script resides on any specific directory, then it needs to set via --base-setup-path
```

Comment:
```suggestion
    - If your client script resides on a different directory, then its path needs to be set via `--base-setup-path`
```
(In general, variable and parameter names should be put between single ticks in markdown)